cmd/olclient/allegation.go

```diff
@@ -0,0 +1,154 @@
+/*
+   ____             _              _                      _____           _                  _
+  / __ \           | |            | |                    |  __ \         | |                | |
+ | |  | |_ __   ___| |     ___  __| | __ _  ___ _ __     | |__) | __ ___ | |_ ___   ___ ___ | |
+ | |  | | '_ \ / _ \ |    / _ \/ _` |/ _` |/ _ \ '__|    |  ___/ '__/ _ \| __/ _ \ / __/ _ \| |
+ | |__| | | | |  __/ |___|  __/ (_| | (_| |  __/ |       | |   | | | (_) | || (_) | (_| (_) | |
+  \____/|_| |_|\___|______\___|\__,_|\__, |\___|_|       |_|   |_|  \___/ \__\___/ \___\___/|_|
+                                      __/ |
+                                     |___/
+
+
+Copyright 2017 - 2019 OneLedger
+*/
+
+package main
+
+import (
+	"path/filepath"
+
+	"github.com/Oneledger/protocol/action"
+	accounts2 "github.com/Oneledger/protocol/data/accounts"
+	"github.com/Oneledger/protocol/data/keys"
+	"github.com/Oneledger/protocol/serialize"
+
+	"github.com/pkg/errors"
+	"github.com/spf13/cobra"
+
+	"github.com/Oneledger/protocol/client"
+	"github.com/Oneledger/protocol/config"
+)
+
+type AllegationArguments struct {
+	Address          []byte `json:"address"`
+	MaliciousAddress []byte `json:"maliciousAddress"`
+	Password         string `json:"password"`
+	BlockHeight      int64  `json:"blockHeight"`
+	ProofMsg         string `json:"proofMsg"`
+}
+
+func (args *AllegationArguments) ClientRequest() client.AllegationRequest {
+	return client.AllegationRequest{
+		Address:          args.Address,
+		BlockHeight:      args.BlockHeight,
+		ProofMsg:         args.ProofMsg,
+		MaliciousAddress: args.MaliciousAddress,
+	}
+}
+
+var allegationCmd = &cobra.Command{
+	Use:   "allegation",
+	Short: "Perform allegation to a validator",
+	RunE:  allegationExec,
+}
+
+// TODO: Move validator signature to staking address and compare if the validator staking address is the same as signed
+
+var allegationArgs = &AllegationArguments{}
+
+func init() {
+	EvidencesCmd.AddCommand(allegationCmd)
+	allegationCmd.Flags().BytesHexVar(&allegationArgs.Address, "address", []byte{}, "address for validator")
+	allegationCmd.Flags().StringVar(&allegationArgs.Password, "password", "", "password to access secure wallet")
+	allegationCmd.Flags().BytesHexVar(&allegationArgs.MaliciousAddress, "maliciousAddress", []byte{}, "malicious validator address")
+	allegationCmd.Flags().Int64Var(&allegationArgs.BlockHeight, "blockHeight", int64(0), "blockHeight for allegation")
+	allegationCmd.Flags().StringVar(&allegationArgs.ProofMsg, "proofMsg", "", "proofMsg for allegation")
+}
+
+func allegationExec(cmd *cobra.Command, args []string) error {
+	ctx := NewContext()
+	ctx.logger.Debug("Have Allegation Request", "allegationArgs", allegationArgs)
+
+	rootPath, err := filepath.Abs(rootArgs.rootDir)
+	if err != nil {
+		return err
+	}
+
+	cfg := &config.Server{}
+
+	//Prompt for password
+	if len(allegationArgs.Password) == 0 {
+		allegationArgs.Password = PromptForPassword()
```

Comment:
@KevinOneLedger @pav-kap Checking password forces validator owner to create a secure wallet account using their private/pub keys before hand. Any problem here?

Reply:
Validator needs the Key Pair for release as well , whatever we decide here should be universal

Reply:
Fine.